### DISCUSSION TITLE
Allow Service Accounts to be assigned Permissions to Dashboards

### DISCUSSION THREAD

**Discussion by MattManjosKodiak** [grafana/grafana#65291] [Dashboards, Panels & Visualizations]
I'm on Grafana 9.4.3 and my workflow involves using hitting `/api/dashboards/uid/${dash}` on all my dashboards in order to pull the json and store it in git, so that I can later re-provision these if I need to rebuild the environment. It's working great and is a nice way to keep that part of grafana ready as IaC.

This script uses a service account with a token to fetch the dashboard json model.

Here's the issue: I have a new dashboard which needs restricted permissions (e.g. the viewer role that the dashboard-fetching service account intentionally can't see it - only editors and admins can load the dashboard).

I'd still like to use the pull-dashboards script to export the json model of this dashboard, without needing to give all other viewer access users on the system the ability to see it.

Is this currently possible? I'd like to assign a service account permissions on individual dashboards. Right now, grafana can only grant permissions to teams, users and roles.

- I don't think I can add a service account to a team
- I can't reference a service account as a user right now
- There's no drop-down in the dashboard permissions window to select service account as an object
- I'd rather not give this service account/api key any permissions beyond viewer

**Comment by MattManjosKodiak** [comment]
Or, maybe hitting this api each time to fetch the dashboards is the wrong tactic and I could be exporting them from sqlite or some other interface?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This is phrased as a question, and we'd like questions about grafana to go to our[ community forum](https://community.grafana.com/). Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
